### PR TITLE
users: Fix crash on nonexisting lastlog

### PR DESCRIPTION
**Pull request by martinpitt**
If `lastlog` is not installed, and neither is `lastlog2`, `getLogins()` ran into an uncaught "not-found" exception due to the spawn failure. The page then never finished loading.

Move the lastlog call closer to where the output is parsed, and handle failure.

----

Spotted in https://github.com/cockpit-project/bots/pull/6491 . This makes the tests mostly work (other than of course the actual "last login" checks)

### CODE REVIEW

**Comment by allisonkarlitskaya** [comment]
Moving the code down to that point separates a comment from the code it was attempting to explain.

Also (suggestion): how about using the empty string as an initializer instead of setting it in the catch block?

File: pkg/users/users.js [hunk 1]
```diff
@@ -216,6 +214,14 @@ async function getLogins(logind_client) {
     }
 
     // drop header and last empty line with slice
+    let lastlog;
```
Comment (allisonkarlitskaya) [inline pkg/users/users.js:217]: `= "";`?

File: pkg/users/users.js [hunk 2]
```diff
@@ -216,6 +214,14 @@ async function getLogins(logind_client) {
     }
 
     // drop header and last empty line with slice
```
Comment (allisonkarlitskaya) [inline pkg/users/users.js:216]: This comment goes with...

File: pkg/users/users.js [hunk 3]
```diff
@@ -216,6 +214,14 @@ async function getLogins(logind_client) {
...
+    let lastlog;
+    try {
+        lastlog = await cockpit.spawn([LastLogPath], { environ: ["LC_ALL=C"] });
+    } catch (ex) {
+        console.warn(`Failed to run ${LastLogPath}: ${ex.toString()}`);
+        lastlog = "";
+    }
+
     const promises = lastlog.split('\n').slice(1, -1).map(async line => {
```
Comment (allisonkarlitskaya) [inline pkg/users/users.js:225]: ...this code.

**Review by allisonkarlitskaya — CHANGES_REQUESTED** [review]

File: pkg/users/users.js (hunk 1 above)
Comment (martinpitt) [reply pkg/users/users.js]: No strong opinion. Changed.

File: pkg/users/users.js (hunk 3 above)
Comment (martinpitt) [reply pkg/users/users.js:224]: Fixed, thanks!

File: pkg/users/users.js [hunk 4]
```diff
@@ -215,6 +213,13 @@ async function getLogins(logind_client) {
         }
     }
 
+    let lastlog = "";
+    try {
+        lastlog = await cockpit.spawn([LastLogPath], { environ: ["LC_ALL=C"] });
+    } catch (ex) {
+        console.warn(`Failed to run ${LastLogPath}: ${ex.toString()}`);
```
Comment (cockpituous) [inline pkg/users/users.js:220]: These 2 added lines are not executed by any test.

**Review by allisonkarlitskaya — APPROVED** [review]

**Pull request merged**